src/status_im2/contexts/chat/lightbox/zoomable_image/utils.cljs

```diff
@@ -0,0 +1,84 @@
+(ns status-im2.contexts.chat.lightbox.zoomable-image.utils
+  (:require
+    [clojure.string :as string]
+    [react-native.orientation :as orientation]
+    [react-native.platform :as platform]
+    [status-im2.contexts.chat.lightbox.zoomable-image.constants :as c]
+    [utils.re-frame :as rf]))
+
+
+(defn get-dimensions
+  "Calculates all required dimensions. Dimensions calculations are different on iOS and Android because landscape
+   mode is implemented differently.On Android, we just need to resize the content, and the OS takes care of the
+   animations. On iOS, we need to animate the content ourselves in code"
+  [pixels-width pixels-height curr-orientation]
+  (let [window                (rf/sub [:dimensions/window])
```

Comment:
@OmarBasem, I'm wondering if memoizing this function could be a safe and sure way of optimizing it because the window dimensions are relatively static, and the rest of the values are mostly derived from the dimensions & orientation (only two possible values). The only thing is double-check if `pixels-width` and `pixels-height` repeat frequently to take advantage of the memoization.

The default memoization function is also known as "naive memoization" because it just keeps growing if new/unique arguments are passed, but if `get-dimensions` arguments are relatively stable, it could give us a nice boost in performance.

The same idea applies to other functions being used in this namespace. Memoizing them could give interesting results given the arguments might not vary much, like `size`, `screen-size`, and `focal`.

Could be an interesting approach if/when you work on further optimizations, not in this PR for sure.

Reply:
Oh and by the way, to make memoization work more reliably, you'd need to remove the subscription `:dimensions/window` and pass the dimensions as arguments.

Reply:
Thanks @ilmotta for your further input. I will try to investigate using memoization in a follow-up PR.